backend/workflow_manager/endpoint/source.py

```diff
@@ -329,7 +329,11 @@ def _is_new_file(self, file_path: str, workflow: Workflow) -> bool:
             workflow=workflow, cache_key=file_hash
         )
 
-        if file_history and file_history.is_completed():
+        if (
+            self.execution_service.use_file_history
+            and file_history
+            and file_history.is_completed()
+        ):
```

Comment:
@chandrasekharan-zipstack could you please explain what exactly are we doing here? May be add a comment in the code also?